### ISSUE TITLE
React ^15.5.0 support?

### ISSUE THREAD

**Issue by ehzhang** [yahoo/react-intl#922]
Hello!

First off, thanks for the great work with this library!

[With the release of React `15.5.0`](https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html), the deprecation of the `react-addons-testutils`and the new `prop-types` package is causes a lot of console errors, which our test suite considers to be test failures.

I was curious if there is current work to make the migration to the relevant packages - I would be happy to also submit a PR for such changes if that's welcome!

Reproduced with a fresh clone of `react-intl`, `npm install && npm test`:

```
 RUNS  test/unit/components/provider.js
 RUNS  test/unit/components/relative.js
 RUNS  test/unit/components/html-message.js
 RUNS  test/unit/components/message.js
 RUNS  test/unit/components/time.js
 RUNS  test/unit/components/date.js

Test Suites: 0 of 16 total
Tests:       0 total
Snapshots:   0 total
  console.error node_modules/fbjs/lib/warning.js:36
    Warning: ReactTestUtils has been moved to react-dom/test-utils. Update references to remove this warning.

  console.error node_modules/fbjs/lib/warning.js:36
    Warning: ReactTestUtils has been moved to react-dom/test-utils. Update references to remove this warning.

  console.error node_modules/fbjs/lib/warning.js:36
    Warning: ReactTestUtils has been moved to react-dom/test-utils. Update references to remove this warning.

  console.error node_modules/fbjs/lib/warning.js:36
 RUNS  test/unit/format.js
 RUNS  test/unit/components/provider.js
 RUNS  test/unit/components/relative.js
 RUNS  test/unit/components/html-message.js
 RUNS  test/unit/components/message.js
 RUNS  test/unit/components/time.js
 RUNS  test/unit/components/date.js

Test Suites: 0 of 16 total
Tests:       0 total
Snapshots:   0 total
  console.error node_modules/fbjs/lib/warning.js:36
    Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.

  console.error node_modules/fbjs/lib/warning.js:36
    Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.
```

**Comment by ghostd** [comment]
Hi,

@inukshuk provided a PR #918 (although it does not replace the 'test-utils' dependency)

**Comment by ehzhang** [comment]
Ah, thanks for bringing that up - somehow missed that!

**Comment by Izopi4a** [comment]
+1

**Comment by terry90** [comment]
+1

**Comment by shai32** [comment]
+1

**Comment by MichaelDeBoey** [comment]
Closed by #918

**Comment by okuryu** [comment]
released v2.3.0.
https://github.com/yahoo/react-intl/releases/tag/v2.3.0
